l2geth/core/state_transition.go

```diff
@@ -122,6 +125,13 @@ func IntrinsicGas(data []byte, contractCreation, isHomestead bool, isEIP2028 boo
 
 // NewStateTransition initialises and returns a new state transition object.
 func NewStateTransition(evm *vm.EVM, msg Message, gp *GasPool) *StateTransition {
+	var l1Fee *big.Int
+	if rcfg.UsingOVM {
+		// Compute the L1 fee before the state transition
+		// so it only has to be read from state one time.
+		l1Fee, _ = fees.CalculateL1MsgFee(msg, evm.StateDB)
```

Comment:
Warning that this will break the expected behavior of eth_call where gasPrice is set to 0 by default. IMO if we're executing a 0 gasPrice transaction then we should also set l1Fee to 0.

Reply:
I was considering this functionality and went back and forth on this. I refactored the integration tests to work with its current implementation, where a 0 gas price transaction would not also trigger a 0 wei L1 fee. I could see a world where we do this and am not super opinionated. One downside of the current approach is that the `gas-oracle` would need to hold value on L2, making `tx.gasPrice = 0` cancel out the L1 portion of the fee would fix this.

This behavior is currently undefined per the spec so would like more opinions @K-Ho @karlfloersch

Reply:
Note that none of the `eth_call` tests are breaking

Reply:
This could be implemented pretty easily by checking `msg.GasPrice` and setting `l1Fee` to `0`

Reply:
I've implemented this and you were right about `eth_call`